components/habits/hooks.ts

```diff
@@ -0,0 +1,55 @@
+import { Entry, HabitWithHistory } from '../../src/types/habits';
+import { CompletionPercentages } from './types';
+
+type EntriesByDay = {
+  [date: string]: Entry[];
+};
+
+const getCompletionPercentage = (quantity: number, target: number) =>
+  (quantity / target) * 100;
+
+/**
+ * Check how much of the target habit has been completed
+ */
+const checkCompletionQuantity = (entries: Entry[]) =>
+  entries.reduce((total, entry) => total + entry.quantity, 0);
+
+// Get the entry for today from the entries history
+const getEntriesForDay = (entries: Entry[], date: string) =>
+  entries.filter(
+    ({ completionDate }) => completionDate.toDateString() === date
+  );
+
+type UseCompletionPercentagesParams = {
+  habitWithHistory: HabitWithHistory;
+  dates: string[];
+};
+
+const useCompletionPercentages = ({
```

Comment:
Does this need to be a hook? I see that there's no state or effects, so actually, it's just a pure function ;) 

Also, you can go with a file per hook, just like you do with components ;)

Reply:
Renamed it to `utils.ts`.